enhancements/installer/dual-stack-enablement.md

```diff
@@ -0,0 +1,271 @@
+
+### Non-Goals
+
+None
+
+### User Stories
+
+- As an Openshift Installer developer, I would like to protect new API fields with a feature 
+gate, so I can build features without concern that an incomplete feature would be exposed to customers.
+
+- As a member of Openshift with customers and/or clients, I would like to explain the benefits of dual stack
+clusters. I want to explain configuring dual stack clusters will allow long term supportability and flexibility
+of our product. This could ensure data and cluster resources are not lost to future compatibility issues as IPv6
+becomes the standard. 
+
+- As a customer, I want to create a cluster that provides access to all possible endpoints now and in the future.
+
+## Proposal
+
+1. Openshift API should include a new variable `StackType` that allows the user to select between
+single stacks and dual stacks.
+
+2. For each cloud provider (AWS, GCP, and Azure), the `platform` section of the install config should allow
+the user to select IPv4_ONLY, IPv6_ONLY, or Dual Stack for the `stackType`. 
+
+3. The cluster api provider for each cloud (AWS, GCP, and Azure) should be updated to allow the correct stack type. 
+
+4. Cluster Manifests need to be altered to include networking details for dual stack clusters.
+
+### Workflow Description
+
+#### Example workflow for feature
+
+1. Developer adds a new enumeration string `stackType` to Openshift API. The new enumeration is tied to features 
+for AWS, GCP, and Azure for the Openshift Installer. The features are set as a Teach Preview Feature Set.
+2. Developer add a new enumeration string `stackType` to the Openshift Installer. The new enumeration should have
+equivalent values to the Openshift API enumeration. This is a Tech Preview feature set feature.
+3. The User may now set the `stackType` in the install configuration file. If the user leaves the `featureSet` field
+blank, then `openshift-install` returns an `error: the TechPreviewNoUpgrade feature set must be enabled to use this field`
+4. Installer validates the values for the enumeration as they must match one of the strings
+5. Installer sets correct subnetwork stack type that is passed to the cluster api provider via the `ClusterSpec` structure
+6. Cluster API Provider sets the correct stack type for the subnetworks in the cluster
+7. The cluster resources are successfully created by the cluster api provider
+8. Installer generates FeatureGate manifest
+9. Installer emits warning message indicating the enabled feature set when provisioning infrastructure
+10. Cluster installs successfully with TechPreview cluster
+
+## Operational Aspects of API Extensions
+
+### API Extensions
+
+#### Openshift
+
+A new feature will be added to the Openshift API for AWS, GCP, and Azure. The feature will make use of the `stackType`
+variable. 
+
+```go
+	// StackType: The stack type for the subnet. If set to IPV4_ONLY, new VMs in
```

Comment:
In OpenShift, it is convention to use the JSON field name in the godoc instead of the Go field name (ref: https://github.com/openshift/enhancements/blob/master/dev-guide/api-conventions.md#use-json-field-names-in-godoc)

This is to make the end-user experience of using something like `oc explain` better by aligning the generated documentation with the field name the end-user sees.